packages/mu-http/driver.js

```diff
@@ -14,4 +14,144 @@
+var assert = require('assert')
+var mue = require('mu-error')()
+
+
+/**
+ * HTTP transport
+ * Figure out path mapping - perhaps this may need to be part of the config for now just /
+ */
+module.exports = function createTcpDriver (options) {
+  return function httpDriver (opts, receive) {
+    var connections = {}
+    var connectionsByIp = {}
+    var server
+
+    assert(opts, 'transport should always pass opts to httpDriver')
+    assert(receive instanceof Function, 'transport should always pass receive function to httpDriver')
+
+
+    function send (message, cb) {
+      var sendTo
+
+      if (connections[message.protocol.dst]) {
+
+        // server responding to client initaiated connection
+        if (connections[message.protocol.dst].length > 0) {
+          sendTo = connections[message.protocol.dst].pop()
+          sendTo.response.writeHead(200, { 'Content-Type': 'application/json' })
+          sendTo.response.write(stringify(message))
+          sendTo.response.end()
+        } else {
+          cb && cb(mue.transport('connection has alredy recieved response!'))
+        }
+      } else {
+
+        // client initiating connection to server
+        var body = []
+        var inbound
+        var req = http.request({host: options.target.host,
+                                port: options.target.port,
+                                method: 'POST',
+                                path: '/',
+                                headers: {'Content-Type': 'application/json'}})
+
+        req.on('response', function (response) {
+          response.on('data', function (chunk) {
+            body.push(chunk)
+          })
+
+          response.on('end', function () {
+            if (response.statusCode !== 200) {
+              receive(mue.transport(response.statusCode + ' ' + response.statusMessage), message)
+              return
+            }
+
+            inbound = parse(Buffer.concat(body))
+            receive(inbound.err, inbound.value)
+          })
+        })
+
+        req.on('error', function (err) {
+          cb(mue.transport(err))
+        })
+
+        req.write(stringify(message))
+        req.end()
+      }
+    }
+
+
+
+    function listen (port, host, ready) {
+      server = http.createServer()
+
+      server.on('request', function (request, response) {
+        var body = []
+        var inbound
+
+        request.on('data', function (chunk) {
+          body.push(chunk)
+        })
+
+        request.on('end', function () {
+          inbound = parse(Buffer.concat(body))
+
+          if (!connections[inbound.value.protocol.src]) {
+            connections[inbound.value.protocol.src] = []
+            connectionsByIp[request.socket.remoteAddress + '_' + request.socket.remotePort] = []
+          }
+          connections[inbound.value.protocol.src].push({request: request, response: response})
+          connectionsByIp[request.socket.remoteAddress + '_' + request.socket.remotePort] = inbound.value.protocol.src
```

Comment:
why are we doing this? an http request is not a socket, and it can't be reuse.

also, we should probably use http://npm.im/end-of-stream to handle all possible cases (including errors)